### PR TITLE
Update dns_dhcp.rst

### PR DESCRIPTION
**Pull request by cotosso**
A phrase fixed and overall improvements

### CODE REVIEW

File: dns_dhcp.rst [hunk 1]
```diff
@@ -73,27 +73,51 @@ This section explains how to configure upstream DNS servers for your system. You
 - Specify a single upstream DNS server: enter the IP address of the desired server in the dedicated field
 - Set up domain-specific DNS servers: this allows you to route queries for specific domains to different servers.
 
-Domain specific DNS Servers:
+Domain specific DNS Servers
```
Comment (filippocarletti) [inline dns_dhcp.rst:76]:
```suggestion
Domain-specific DNS Servers
```

File: dns_dhcp.rst [hunk 2]
```diff
@@ -73,27 +73,51 @@ This section explains how to configure upstream DNS servers for your system. You
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-- Empty domain (``//``): matches unqualified names (without dots).
-- Specific domain (``/google.com/``): matches the exact domain and its subdomains (e.g., google.com, www.google.com).
-- Wildcard domain (``*google.com/``): matches any domain containing "google.com" (e.g., google.com, www.google.com, supergoogle.com).
-- Non-wildcard takes priority: if both specific and wildcard domains are defined for the same pattern, the specific one takes precedence (e.g., ``/google.com/`` will handle google.com and www.google.com, leaving supergoogle.com to the wildcard).
+The syntax for indicating a DNS server for a specific domain is as follows:
```
Comment (filippocarletti) [inline dns_dhcp.rst:79]:
```suggestion
To use a custom DNS server for a specific domain, use the following syntax:
```

File: dns_dhcp.rst [hunk 3]
```diff
@@ -73,27 +73,51 @@ This section explains how to configure upstream DNS servers for your system. You
...
+
+``/DOMAIN_OPTION/IP_ADDRESS#PORT``
+
+where:
+
+- IP_ADDRESS : specify the IP address of the desired server.
```
Comment (filippocarletti) [inline dns_dhcp.rst:85]:
```suggestion
- IP_ADDRESS: specify the IP address of the desired server
```

File: dns_dhcp.rst [hunk 4]
```diff
@@ -73,27 +73,51 @@ This section explains how to configure upstream DNS servers for your system. You
...
+- PORT: append the desired port (after the IP address using `#` character).
+
+The ``PORT`` value is optional so usually the configuration appears just like:
 
-Server address:
+``/DOMAIN_OPTION/IP_ADDRESS``
```
Comment (filippocarletti) [inline dns_dhcp.rst:90]: Maybe /DOMAIN/IP_ADDRESS#PORT

File: dns_dhcp.rst [hunk 5]
```diff
@@ -73,27 +73,51 @@ This section explains how to configure upstream DNS servers for your system. You
...
 
-- IP address: specify the IP address of the desired server.
-- Port: append the desired port after the IP address using `#`.
+These are the main supported options.
```
Comment (filippocarletti) [inline dns_dhcp.rst:92]:
```suggestion
These are the main supported options:
```

File: dns_dhcp.rst [hunk 6]
```diff
@@ -73,27 +73,51 @@ This section explains how to configure upstream DNS servers for your system. You
...
 
+
+Conflicts management
+
+More specific domains take precedence over less specific domains, so for a configuration like this:
+
+- ``/google.com/1.2.3.4``
+- ``/www.google.com/2.3.4.5``
+
+NethSecurity will send queries  for google.com and gmail.google.com to 1.2.3.4, but www.google.com will go to 2.3.4.5
```
Comment (filippocarletti) [inline dns_dhcp.rst:113]:
```suggestion
NethSecurity will send queries for google.com and gmail.google.com to 1.2.3.4, but www.google.com will go to 2.3.4.5
```

File: dns_dhcp.rst [hunk 7]
```diff
@@ -73,27 +73,51 @@ This section explains how to configure upstream DNS servers for your system. You
...
+
+This is true also for wildcard: if both specific and wildcard domains are defined for the same pattern, the specific one takes precedence (e.g., having ``/google.com/`` and ``/*google.com/`` : the first will handle google.com and www.google.com, the wildcard will handle supergoogle.com.
```
Comment (filippocarletti) [inline dns_dhcp.rst:115]:
```suggestion
This is true also for wildcards: if both specific and wildcard domains are defined for the same pattern, the specific one takes precedence (e.g., having ``/google.com/`` and ``/*google.com/`` : the first will handle google.com and www.google.com, the wildcard will handle supergoogle.com.
```

**Review by filippocarletti — CHANGES_REQUESTED** [review]

File: dns_dhcp.rst [hunk 8]
```diff
@@ -73,27 +73,51 @@ This section explains how to configure upstream DNS servers for your system. You
...
+- ``/google.com/1.2.3.4``
+- ``/www.google.com/2.3.4.5``
+
+NethSecurity will send queries for google.com and gmail.google.com to 1.2.3.4, but www.google.com will go to 2.3.4.5
+
+This is true also for wildcards: if both specific and wildcard domains are defined for the same pattern, the specific one takes precedence (e.g., having ``/google.com/`` and ``/*google.com/`` : the first will handle google.com and www.google.com, the wildcard will handle supergoogle.com.
+
+
+
+
```
Comment (gsanchietti) [inline dns_dhcp.rst:119]:
```suggestion
```

File: dns_dhcp.rst [hunk 9]
```diff
@@ -73,27 +73,47 @@ This section explains how to configure upstream DNS servers for your system. You
...
-  ``/google.com/1.2.3.4``
+- Send all queries for "google.com" and its subdomains to 1.2.3.4:  ``/google.com/1.2.3.4``
 - Send all unqualified names (e.g., "localhost") to 10.0.0.1 and everything else to standard servers: ``//10.0.0.1``
-
-- Send queries for "internal.nethserver.org" and its subdomains to 192.168.1.1 and everything else to standard servers:
+- Send queries for domain "ad.nethserver.org" and its subdomains to 192.168.1.1 and everything else to standard servers:
   ``/ad.nethserver.org/192.168.1.1``
 
+
+Conflicts management
```
Comment (gsanchietti) [inline dns_dhcp.rst:106]:
```suggestion
```

File: dns_dhcp.rst [hunk 10]
```diff
@@ -73,27 +73,45 @@ This section explains how to configure upstream DNS servers for your system. You
...
+Domain-specific DNS Servers
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-- Empty domain (``//``): matches unqualified names (without dots).
-- Specific domain (``/google.com/``): matches the exact domain and its subdomains (e.g., google.com, www.google.com).
-- Wildcard domain (``*google.com/``): matches any domain containing "google.com" (e.g., google.com, www.google.com, supergoogle.com).
-- Non-wildcard takes priority: if both specific and wildcard domains are defined for the same pattern, the specific one takes precedence (e.g., ``/google.com/`` will handle google.com and www.google.com, leaving supergoogle.com to the wildcard).
+To use a custom DNS server for a specific domain, use the following syntax:
+
+``/DOMAIN_OPTION/IP_ADDRESS#PORT``
```
Comment (gsanchietti) [inline dns_dhcp.rst:81]:
```suggestion
``/DOMAIN/IP_ADDRESS#PORT``
```

File: dns_dhcp.rst [hunk 11]
```diff
@@ -73,27 +73,45 @@ This section explains how to configure upstream DNS servers for your system. You
...
+
+- IP_ADDRESS: specify the IP address of the desired server
+- PORT: append the desired port (after the IP address using `#` character).
 
-Server address:
+The ``PORT`` value is optional so usually the configuration appears just like:
 
-- IP address: specify the IP address of the desired server.
-- Port: append the desired port after the IP address using `#`.
+``/DOMAIN_OPTION/IP_ADDRESS``
```
Comment (gsanchietti) [inline dns_dhcp.rst:90]:
```suggestion
``/DOMAIN/IP_ADDRESS``
```

**Review by gsanchietti — APPROVED** [review]

**Pull request merged**